core/src/eval/mod.rs

```diff
@@ -938,21 +938,23 @@ impl<R: ImportResolver, C: Cache> VirtualMachine<R, C> {
 
     /// Evaluate a term, but attempt to continue on errors.
     ///
-    /// This differs from `VirtualMachine::eval_full` in 3 ways:
+    /// This differs from `VirtualMachine::eval_full` in 2 ways:
     /// - We try to accumulate errors instead of bailing out. When recursing into record
     ///   fields and array elements, we keep evaluating subsequent elements even if one
     ///   fails.
-    /// - We ignore missing field errors. It would be nice not to ignore them, but it's hard
-    ///   to tell when they're appropriate: the term might intentionally be a partial configuration.
     /// - We only return the accumulated errors; we don't return the eval'ed term.
     pub fn eval_permissive(&mut self, rt: RichTerm) -> Vec<EvalError> {
         fn inner<R: ImportResolver, C: Cache>(
             slf: &mut VirtualMachine<R, C>,
             acc: &mut Vec<EvalError>,
             rt: RichTerm,
         ) {
+            let pos = rt.pos;
             match slf.eval(rt) {
-                Err(e) => acc.push(e),
+                Err(e) => {
+                    acc.push(e);
+                    slf.stack.reset(&mut slf.cache);
```

Comment:
I think `VirtualMachine` has a reset function. Mostly just call `stack.reset()` and `call_stack.clear()`, but you don't need to access inner fields of the virtual machine.